sdk/communication/azure-communication-messages/samples/get_templates_list_async.py

```diff
@@ -0,0 +1,55 @@
+# coding: utf-8
+
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+FILE: get_templates_list_async.py
+
+DESCRIPTION:
+    This sample demonstrates fetching WhatsApp templates created in your WhatsApp Business account. The NotificationMessageClient is 
+    authenticated using a connection string.
+USAGE:
+    python get_templates_list_async.py
+
+    Set the environment variable with your own value before running the sample:
+    1) COMMUNICATION_SAMPLES_CONNECTION_STRING - the connection string in your ACS resource
+    2) WHATSAPP_CHANNEL_ID - Channel created in Azure portal for Advanced Messaging.
+"""
+
+import asyncio
+import os
+import sys
+
+sys.path.append("..")
+
+class GetTemplatesSampleAsync(object):
+
+    connection_string = os.getenv("COMMUNICATION_SAMPLES_CONNECTION_STRING")
+    channel_id = os.getenv("WHATSAPP_CHANNEL_ID")
+    
+    async def get_templates_list_async(self):
+
+        from azure.communication.messages.aio import MessageTemplateClient
+
+        message_template_client = MessageTemplateClient.from_connection_string(self.connection_string)
+        
+        # calling send() with whatsapp message details
+        async with message_template_client:
+            template_list = message_template_client.list_templates(self.channel_id)
+            async_list_data = [x async for x in template_list]
+            if (template_list is not None):
```

Comment:
`template_list` will never be None. And if the operation failed, an error would be raised. You can replace this entire `if/else` statement with just:
```python
templates = [x async for x in template_list]
print(f"Successfully retrieved {len(templates)} templates from channel_id {self.channel_id}.")
```